packages/tests/tests/peer_exchange.node.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { bootstrap } from "@libp2p/bootstrap";
+import { waitForRemotePeer } from "@waku/core";
+import {
+  Fleet,
+  getPredefinedBootstrapNodes,
+} from "@waku/core/lib/predefined_bootstrap_nodes";
+import { createFullNode } from "@waku/create";
+import type { PeerExchangeResponse, WakuFull } from "@waku/interfaces";
+import { Protocols } from "@waku/interfaces";
+import { expect } from "chai";
+
+import { delay } from "../src/delay.js";
+import { makeLogFileName, Nwaku } from "../src/index.js";
+
+describe("Peer Exchange: Node", () => {
+  let waku: WakuFull;
+  let nwaku1: Nwaku;
+  let nwaku2: Nwaku;
+  let nwaku3: Nwaku;
+  afterEach(async function () {
+    !!nwaku1 && nwaku1.stop();
+    !!nwaku2 && nwaku2.stop();
+    !!nwaku3 && nwaku3.stop();
+
+    !!waku && waku.stop().catch((e) => console.log("Waku failed to stop", e));
+  });
+
+  // skipping in CI as this test demonstrates Peer Exchange working with the test fleet
+  // but not with locally run nwaku nodes
+  it.skip("Locally run nwaku nodes: Queries successfully", async function () {
```

Comment:
If the test fails because we are still pending some investigation then just delete it, you can have a different branch that just introduces the test so that you and @kaiserd can investigate but no need to push it to master.